cli/command/trust/signer_remove.go

```diff
@@ -0,0 +1,129 @@
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-remove NAME IMAGE [IMAGE...]",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.signer = args[0]
+			options.images = args[1:]
+			return removeSigner(dockerCli, &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, options *signerRemoveOptions) error {
+	if len(options.images) < 1 {
+		return fmt.Errorf("at least one image must be provided")
+	}
+
+	for _, image := range options.images {
+		if err := removeSingleSigner(cli, image, options.signer, options.forceYes); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isLastSigner(role data.Role, allRoles []client.RoleWithSignatures) bool {
+	var lastSigner client.RoleWithSignatures
+	for i := range allRoles {
+		if allRoles[i].Name == "targets/releases" {
+			lastSigner = allRoles[i]
+		}
+	}
+	if len(lastSigner.Signatures) == 0 {
+		fmt.Printf("All signed tags are currently revoked, use docker trust sign to fix")
+		return false
+	}
+	for _, signature := range lastSigner.Signatures {
+		for _, key := range role.KeyIDs {
+			if signature.KeyID != key {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func removeSingleSigner(cli command.Cli, image, signerName string, forceYes bool) error {
+	_, ref, repoInfo, authConfig, err := getImageReferencesAndAuth(cli, image)
+	if err != nil {
+		return err
+	}
+	delegation := "targets/" + signerName
+	var role data.Role
+	if delegation == releases {
+		return fmt.Errorf("releases is a reserved keyword and cannot be removed")
+	}
+	notaryRepo, err := trust.GetNotaryRepository(cli, repoInfo, *authConfig, "push", "pull")
+	if err != nil {
+		return trust.NotaryError(ref.Name(), err)
+	}
+	delegationRoles, err := notaryRepo.GetDelegationRoles()
+	if err != nil {
+		return fmt.Errorf("Error retrieving signers for %s", image)
+	}
+	for _, delRole := range delegationRoles {
+		if delRole.Name == data.RoleName(delegation) {
+			role = delRole
+		}
+	}
+	if role.Name == "" {
+		return fmt.Errorf("No  signer %s for image %s", signerName, image)
+	}
+	allRoles, err := notaryRepo.ListRoles()
+	if err != nil {
+		return err
+	}
+	if ok := isLastSigner(role, allRoles); ok && !forceYes {
+		removeSigner := command.PromptForConfirmation(os.Stdin, cli.Out(), fmt.Sprintf("The signer %s signed the last released version of %s. "+
+			"Removing them will lead to %s being unpullable."+
+			"Are you sure you want to continue? (y/n): ",
+			signerName, image, image,
+		))
+
+		if !removeSigner {
+			fmt.Fprintf(cli.Out(), "\nAborting action. \n")
+			return nil
+		}
+	} else if err != nil {
+		return err
+	}
+	if err = notaryRepo.RemoveDelegationKeys(releases, role.KeyIDs); err != nil {
```

Comment:
can we have a test for what happens when the signer's key is not in `targets/releases`, but they have their own role still?

Also: if the signer is neither in `targets/releases` nor `targets/<signer>`?

I'm not sure if we can do much here without a manual test for now.  It would be good to track these tests in the existing issue or a new one.